frontend/src/pages/MetricsApplication/ApDex/ApDexSettings.tsx

```diff
@@ -0,0 +1,124 @@
+import { CloseOutlined, QuestionCircleOutlined } from '@ant-design/icons';
+import { Card, Input } from 'antd';
+import { AxiosResponse } from 'axios';
+import Spinner from 'components/Spinner';
+import { themeColors } from 'constants/theme';
+import { useSetApDexSettings } from 'hooks/apDex/useSetApDexSettings';
+import { useNotifications } from 'hooks/useNotifications';
+import { useEffect, useState } from 'react';
+import { ApDexPayloadProps } from 'types/api/metrics/getApDex';
+
+import {
+	AppDexThresholdContainer,
+	Button,
+	ExcludeErrorCodeContainer,
+	SaveAndCancelContainer,
+	SaveButton,
+	Typography,
+} from '../styles';
+import { onSaveApDexSettings } from '../utils';
+
+function ApDexSettings({
+	servicename,
+	handlePopOverClose,
+	isLoading,
+	data,
+	refetch,
+}: ApDexSettingsProps): JSX.Element {
+	const [thresholdValue, setThresholdValue] = useState(0);
+	const { notifications } = useNotifications();
+
+	const { isLoading: setApDexIsLoading, mutateAsync } = useSetApDexSettings({
+		servicename,
+		threshold: thresholdValue,
+		excludeStatusCode: '',
+	});
+
+	useEffect(() => {
```

Comment:
why useEffect here can use the initial callback of useState ?

Reply:
Okay.